aws/cloudfront_distribution_configuration_structure.go

```diff
@@ -763,15 +606,15 @@ func originHash(v interface{}) int {
 		buf.WriteString(fmt.Sprintf("%d-", customHeadersHash(v.(*schema.Set))))
 	}
 	if v, ok := m["custom_origin_config"]; ok {
-		if s := v.(*schema.Set).List(); len(s) > 0 {
+		if s := v.([]interface{}); len(s) > 0 && s[0] != nil {
```

Comment:
`nil` checks specifically added here as I was able to coerce a panic during testing.